vpr/src/route/rr_graph_generation/rr_graph_opin_chan_edges.cpp

```diff
@@ -0,0 +1,721 @@
+
+#include "rr_graph_opin_chan_edges.h"
+
+#include "rr_graph2.h"
+#include "rr_graph_sg.h"
+#include "rr_graph_builder.h"
+#include "rr_graph_view.h"
+#include "rr_types.h"
+#include "rr_graph_type.h"
+#include "clb2clb_directs.h"
+#include "get_parallel_segs.h"
+#include "build_scatter_gathers.h"
+
+#include "globals.h"
+#include "physical_types_util.h"
+
```

Comment:
Add Doxygen comment for what should go in this file. 
Add Doxygen comments for any routines where you know what they do in this file.